arm-recoveryservicesbackup/2016-06-01/swagger/recoveryservicesbackup.json

```diff
@@ -0,0 +1,1548 @@
+            "in": "path",
+            "description": "OperationID which represents the operation",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "204": {
+            "description": "NoContent"
+          }
+        },
+        "deprecated": false
+      }
+    }
+  },
+  "definitions": {
+    "Object": {
+      "description": "Base of all objects.",
+      "type": "object",
+      "properties": {}
+    },
+    "OperationStatus": {
+      "description": "Operation status.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "ID of the operation.",
+          "type": "string"
+        },
+        "name": {
+          "description": "Name of the operation.",
+          "type": "string"
+        },
+        "status": {
+          "description": "Operation status. Possible values: InProgress, Succeeded, Failed, Canceled.",
+          "type": "string"
+        },
+        "startTime": {
+          "format": "date-time",
+          "description": "Operation start time.",
+          "type": "string"
+        },
+        "endTime": {
+          "format": "date-time",
+          "description": "Operation end time.",
+          "type": "string"
+        },
+        "error": {
+          "$ref": "#/definitions/OperationStatusError",
+          "description": "Error information related to this operation."
+        },
+        "properties": {
+          "$ref": "#/definitions/OperationStatusExtendedInfo",
+          "description": "Additional information associated with this operation."
+        }
+      }
+    },
+    "OperationStatusError": {
+      "description": "Error information associated with operation status call.",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "Error code of the operation failure.",
+          "type": "string"
+        },
+        "message": {
+          "description": "Error message displayed if the operation failure.",
+          "type": "string"
+        }
+      }
+    },
+    "OperationStatusExtendedInfo": {
+      "description": "Base class for additional information of operation status.",
+      "type": "object",
+      "properties": {}
+    },
+    "ResourceList[Job]": {
+      "description": "Resource list wrapper for \"T\" type",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Gets or sets the list of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Resource[Job]"
+          }
+        },
+        "nextLink": {
+          "description": "Gets or sets the uri to fetch the next page of resources. Call ListNext() fetches next page of resources.",
+          "type": "string"
+        }
+      }
+    },
+    "Resource[Job]": {
```

Comment:
Rather than putting the type information in the name (i.e. `Resource[Job]`), I would recommend changing the definitions structure a bit:

You could have a base `Resource` type that has all of the Resource properties (e.g. `id`, `name`). Please include the [`x-ms-azure-resource` extension] to make sure that this inherits from the `Resource` type from the client runtime. It helps customers, since `Resource` is common across all services.

`ProtectedItem` and `Job` definitions can then inherit from the `Resource` definition using JSON allOf: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#model-inheritance. That way `ProtectedItem` and `Job` have the Resource properties and don't re-define them.